### PR TITLE
Login

### PR DESCRIPTION
**Pull request by AlexisN94**
- Show Login button in Profile fragment if not logged in
- Store token in Encrypted Shared Preferences
- Save user info in database

### CODE REVIEW

File: app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    fun setToken(domainToken: DomainToken) {
+        val tokenJson = jsonAdapter.toJson(domainToken)
+        sharedPreferences.edit()
+            .putString("token", tokenJson)
+            .apply()
+    }
+
+    fun getToken(): DomainToken? {
+        if (!hasToken())
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt]: use curly brackets or single line if

File: app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    private val jsonAdapter: JsonAdapter<DomainToken> = moshi.adapter(DomainToken::class.java)
+
+    fun setToken(domainToken: DomainToken) {
+        val tokenJson = jsonAdapter.toJson(domainToken)
+        sharedPreferences.edit()
+            .putString("token", tokenJson)
+            .apply()
+    }
+
+    fun getToken(): DomainToken? {
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt]: Consider renaming this as fetchToken() to indicate that is not just returning value from a variable token

File: app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+
+private const val MIN_MILLIS_REMAINING_ALLOWED = 60 * 1000 // 1 min
+
+class TokenStorageManager private constructor() {
+    private lateinit var sharedPreferences: EncryptedSharedPreferences
+    private val moshi: Moshi = Moshi.Builder().add(KotlinJsonAdapterFactory()).build()
+    private val jsonAdapter: JsonAdapter<DomainToken> = moshi.adapter(DomainToken::class.java)
+
+    fun setToken(domainToken: DomainToken) {
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt]: Consider renaming this method as updateToken or something similar

File: app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        val tokenJson = jsonAdapter.toJson(domainToken)
+        sharedPreferences.edit()
+            .putString("token", tokenJson)
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt]: Can this value be named token_json instead or something similar? jsonToken is a better name for the variable :P

Comment (andre-exd) [reply app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt]: Consider creating a private constant for field on SharedPreferences to be reused on other places

File: app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            .apply()
+    }
+
+    fun getToken(): DomainToken? {
+        if (!hasToken())
+            return null
+
+        val tokenJson = sharedPreferences.getString("token", null)
+        return jsonAdapter.fromJson(tokenJson)
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt:28]: Can you handle null from tokenJson more explicitly? probably returning a explicit null is better then returning a null from jsonAdapter.fromJson(). Not sure if an exception will happen if tokenJson is null from...

File: app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    fun checkExpired(): Boolean {
+        val token = getToken()
+        return checkExpired(token)
+    }
+
+    @Suppress("MemberVisibilityCanBePrivate")
+    fun checkExpired(domainToken: DomainToken?): Boolean {
+        if (domainToken == null)
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt]: curly brackets or single line if

File: app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt (hunk 1 above)
Comment (andre-exd) [reply app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt]: Please remove hasToken() and handle null result from sharedPreferences instead. That was only one access to SharedPreferences will happen and you will remove adding methods that are only be used once

File: app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt [hunk 7]
```diff
@@ -2,40 +2,68 @@ package com.alexis.myanimecompanion.data
...
 
-    suspend fun updateAnimeStatus(anime: Anime?) {
+    suspend fun updateAnimeStatus(anime: Anime) {
         /* TODO Save changes to localDataSource and remoteDataSource if applicable */
     }
 
-    suspend fun getAnime(anime: Anime?): Anime? {
-        /* TODO return complete? Anime object with fresh user-specific status */
+    suspend fun getAnime(anime: Anime): Anime? {
+        /* TODO return complete? Anime object with fresh domainUser-specific status */
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt]: Was this caused by a search and replace? Please revert it please

File: app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt [hunk 8]
```diff
@@ -2,40 +2,68 @@ package com.alexis.myanimecompanion.data
...
-        /* TODO return complete? Anime object with fresh user-specific status */
+        /* TODO return complete? Anime object with fresh domainUser-specific status */
         return remoteDataSource.getAnimeDetails(animeId)
     }
 
+    suspend fun getUser(): DomainUser? {
+        val remoteUserAsDatabaseUser = remoteDataSource.getUser()
+        val localUser = localDataSource.getUser()
+
+        if(localUser == null && remoteUserAsDatabaseUser != null) {
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt]: Fix formatting

File: app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt [hunk 9]
```diff
@@ -2,40 +2,68 @@ package com.alexis.myanimecompanion.data
...
         return remoteDataSource.getAnimeDetails(anime)
     }
 
     suspend fun getAnime(animeId: Int): Anime? {
-        /* TODO return complete? Anime object with fresh user-specific status */
+        /* TODO return complete? Anime object with fresh domainUser-specific status */
         return remoteDataSource.getAnimeDetails(animeId)
     }
 
+    suspend fun getUser(): DomainUser? {
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt]: Rename method to fetchUser()

File: app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt [hunk 10]
```diff
@@ -2,40 +2,68 @@ package com.alexis.myanimecompanion.data
...
+        val remoteUserAsDatabaseUser = remoteDataSource.getUser()
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt]: Is it really necessary to name this as "...AsDatabase..."? Please simplify if you agree

File: app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt (hunk 9 above)
Comment (andre-exd) [reply app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt]: Or better, name something that indicates that Database will be updated

File: app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt [hunk 11]
```diff
@@ -2,40 +2,68 @@ package com.alexis.myanimecompanion.data
...
+
+        return localDataSource.getUser()
+    }
+
+    fun logout() {
+        localDataSource.clearUser()
+        remoteDataSource.clearUser()
+    }
+
+    fun getAuthorizationUrl(): String {
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt]: Either this method or datasource should be renamed. Please make it consistent.

File: app/src/main/java/com/alexis/myanimecompanion/data/LocalDataSource.kt [hunk 12]
```diff
@@ -2,10 +2,30 @@ package com.alexis.myanimecompanion.data
...
+import com.alexis.myanimecompanion.domain.DomainUser
 
 class LocalDataSource private constructor() {
     private lateinit var animeDatabase: AnimeDatabase
 
+    fun insertUser(user: DatabaseUser){
+        animeDatabase.userDao.insert(user)
+    }
+
+    fun getUser(): DomainUser? {
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/LocalDataSource.kt]: I believe this is not a responsibility of LocalDataSource. Please consider changing the result to DatabaseUser as the rest of the methods.

File: app/src/main/java/com/alexis/myanimecompanion/data/LocalDataSource.kt [hunk 13]
```diff
@@ -2,10 +2,30 @@ package com.alexis.myanimecompanion.data
...
+        return animeDatabase.userDao.getUser()?.asDomainUser()
+    }
+
+    fun clearUser() {
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/LocalDataSource.kt]: Consider using deleteUser(user: DatabaseUser) instead

File: app/src/main/java/com/alexis/myanimecompanion/data/local/daos/UserDao.kt [hunk 14]
```diff
@@ -14,6 +14,6 @@ interface UserDao {
     @Delete
     fun delete(databaseUser: DatabaseUser)
 
-    @Query("SELECT * FROM DatabaseUser WHERE id = :userId")
-    fun getUserById(userId: Int) : DatabaseUser
+    @Query("SELECT * FROM DatabaseUser")
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/local/daos/UserDao.kt]: This doesn't seem a good approach. Please consider reverting to previous query and store userId in sharedPreferences if needed

File: app/src/main/java/com/alexis/myanimecompanion/data/remote/models/AnimeStatistics.kt [hunk 15]
```diff
@@ -1,7 +1,7 @@
 package com.alexis.myanimecompanion.data.remote.models
 
 /**
- * General anime statistics for the user
+ * General anime statistics for the domainUser
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/remote/models/AnimeStatistics.kt]: Search and replace?

File: app/src/main/java/com/alexis/myanimecompanion/domain/DomainToken.kt [hunk 16]
```diff
@@ -0,0 +1,8 @@
+package com.alexis.myanimecompanion.domain
+
+data class DomainToken(
+    var accessToken: String,
+    var expiresAt: Long,
+    var refreshToken: String,
+    var tokenType: String
+)
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/domain/DomainToken.kt]: EOF

File: app/src/main/java/com/alexis/myanimecompanion/domain/DomainUser.kt [hunk 17]
```diff
@@ -0,0 +1,8 @@
+package com.alexis.myanimecompanion.domain
+
+data class DomainUser(
+    val id: Int = -1,
+    val username: String? = null,
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/domain/DomainUser.kt:5]: Can you remove the default values?

File: app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileFragment.kt [hunk 18]
```diff
@@ -7,12 +7,42 @@ import android.view.LayoutInflater
...
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        TODO("If user is logged in, proceed. Otherwise, offer button to redirect to login fragment")
-        return super.onCreateView(inflater, container, savedInstanceState)
+        val binding = FragmentProfileBinding.inflate(inflater)
+        val animeRepository = AnimeRepository.getInstance(requireContext())
+
+        val viewModelFactory = ProfileViewModelFactory(animeRepository)
+        viewModel = ViewModelProvider(viewModelStore, viewModelFactory)[ProfileViewModel::class.java]
+
+        binding.lifecycleOwner = viewLifecycleOwner
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileFragment.kt]: Try to reuse binding.apply {} as on other Fragments

File: app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileFragment.kt [hunk 19]
```diff
@@ -7,12 +7,42 @@ import android.view.LayoutInflater
...
+    }
+
+    private fun onStartLogin() {
+        val authorizationUrl = viewModel.getAuthorizationUrl()
+        val browse = Intent(Intent.ACTION_VIEW, Uri.parse(authorizationUrl))
+        startActivity(browse)
+        viewModel.onStartLoginHandled()
     }
 }
+
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileFragment.kt]: Remove extra lines

File: app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileViewModel.kt [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+        get() = _user
+
+    private val _evtStartLogin = MutableLiveData<Boolean>()
+    val evtStartLogin: LiveData<Boolean>
+        get() = _evtStartLogin
+
+    init {
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                _loading.postValue(true)
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileViewModel.kt]: Consider using a call to updatedLoading(true)

File: app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileViewModel.kt [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+                getUser()
+                _loading.postValue(false)
+            }
+        }
+    }
+
+    private suspend fun getUser() {
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileViewModel.kt]: Rename method to not created confusion with a getter to user value

File: app/src/main/res/layout/fragment_profile.xml [hunk 22]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
```
Comment (andre-exd) [inline app/src/main/res/layout/fragment_profile.xml]: tools:context

File: app/src/main/res/layout/fragment_profile.xml [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <androidx.constraintlayout.widget.Group
+            android:id="@+id/group_not_logged_in"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:visibility="@{(viewModel.user == null &amp;&amp; !viewModel.loading) ? View.VISIBLE : View.GONE}"
```
Comment (andre-exd) [inline app/src/main/res/layout/fragment_profile.xml]: Can't you use a BindingAdapter to transform a boolean into VISIBLE or GONE?

File: app/src/main/res/layout/fragment_profile.xml [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintHorizontal_bias="0.494"
+            app:layout_constraintStart_toStartOf="parent" />
+
+        <androidx.appcompat.widget.AppCompatButton
+            android:id="@+id/btn_profile_login"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:onClick="@{()->viewModel.onStartLogin()}"
+            android:text="Login"
```
Comment (andre-exd) [inline app/src/main/res/layout/fragment_profile.xml]: strings.xml?

File: app/src/main/res/layout/fragment_profile.xml [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+            android:id="@+id/tv_profile_username"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@{`Username: ` + viewModel.user.username}"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+    </androidx.constraintlayout.widget.ConstraintLayout>
+</layout>
```
Comment (andre-exd) [inline app/src/main/res/layout/fragment_profile.xml]: EOF

File: app/src/main/res/navigation/nav_graph.xml [hunk 26]
```diff
@@ -56,13 +56,5 @@
...
-        <action
-            android:id="@+id/action_profileFragment_to_loginFragment"
-            app:destination="@id/loginFragment" />
-    </fragment>
-    <fragment
-        android:id="@+id/loginFragment"
-        android:name="com.alexis.myanimecompanion.ui.login.LoginFragment"
-        android:label="LoginFragment" />
+        android:label="ProfileFragment" />
 </navigation>
```
Comment (andre-exd) [inline app/src/main/res/navigation/nav_graph.xml]: Consider adding a new line at end of line. Not sure if file might be regenerated and that change be lost...

**Review by andre-exd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+        return millisLeft <= MIN_MILLIS_REMAINING_ALLOWED
+    }
+
+    fun clearToken() {
+        sharedPreferences.edit().clear()
+    }
+
+    companion object {
+        private var INSTANCE: TokenStorageManager? = null
+
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/TokenStorageManager.kt:50]: Remove extra line

File: app/src/main/java/com/alexis/myanimecompanion/Utils.kt [hunk 28]
```diff
@@ -91,3 +93,17 @@ fun Activity.dismissKeyboard() {
     if (inputMethodManager.isAcceptingText)
         inputMethodManager.hideSoftInputFromWindow(currentFocus?.windowToken, 0)
 }
+
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/Utils.kt:96]: Remove extra line

File: app/src/main/java/com/alexis/myanimecompanion/Utils.kt [hunk 29]
```diff
@@ -91,3 +93,17 @@ fun Activity.dismissKeyboard() {
...
+    return EncryptedSharedPreferences
+        .create(
+            fileName,
+            masterKeyAlias,
+            context.applicationContext,
+            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+        ) as EncryptedSharedPreferences
+}
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/Utils.kt]: Please re-add new line at EOF

File: app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt [hunk 30]
```diff
@@ -3,17 +3,20 @@ package com.alexis.myanimecompanion.data
 import android.content.Context
 import com.alexis.myanimecompanion.domain.Anime
 
+
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/AnimeRepository.kt]: Remove extra line

File: app/src/main/java/com/alexis/myanimecompanion/data/UserRepository.kt [hunk 31]
```diff
@@ -0,0 +1,78 @@
...
+                        instance.remoteDataSource = RemoteDataSource.getInstance(context)
+
+                        instance.sharedPreferences = createEncryptedSharedPreferences(context, USER_SP_FILENAME)
+
+                        INSTANCE = instance
+                    }
+            }
+        }
+    }
+}
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/data/UserRepository.kt]: EOF

File: app/src/main/java/com/alexis/myanimecompanion/ui/BindingAdapters.kt [hunk 32]
```diff
@@ -108,3 +104,20 @@ fun AppCompatEditText.setValueListener(listener: InverseBindingListener) {
...
+}
+
+@BindingAdapter("showIfNotLoggedIn")
+fun Group.showIfNotLoggedIn(viewModel: ProfileViewModel) {
+    visibility = if (viewModel.user.value == null && !viewModel.loading.value!!) {
+        View.VISIBLE
+    } else {
+        View.GONE
+    }
+}
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/ui/BindingAdapters.kt]: EOF

File: app/src/main/java/com/alexis/myanimecompanion/ui/BindingAdapters.kt [hunk 33]
```diff
@@ -108,3 +104,20 @@ fun AppCompatEditText.setValueListener(listener: InverseBindingListener) {
     }
 }
 
+@BindingAdapter("showIfLoggedIn")
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/ui/BindingAdapters.kt]: Can you change this BindingAdapter to something more generic that accepts a boolean and converts it into a visibility? Something like "visible(boolean: isVisible)"

File: app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileViewModel.kt [hunk 34]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    fun onStartLoginHandled() {
+        _evtStartLogin.value = false
+    }
+
+    fun getAuthorizationUrl(): String {
+        return userRepository.getAuthorizationUrl()
+    }
+}
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileViewModel.kt]: EOF

File: app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileViewModelFactory.kt [hunk 35]
```diff
@@ -0,0 +1,19 @@
...
+    private val userRepository: UserRepository
+) : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(ProfileViewModel::class.java)) {
+            return ProfileViewModel(animeRepository, userRepository) as T
+        }
+        throw IllegalArgumentException("Unknown ViewModel class")
+    }
+}
+
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileViewModelFactory.kt]: Remove extra line

**Review by andre-exd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/alexis/myanimecompanion/ui/BindingAdapters.kt (hunk 32 above)
Comment (AlexisN94) [reply app/src/main/java/com/alexis/myanimecompanion/ui/BindingAdapters.kt:123]: was fixed in base branch

File: app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileViewModel.kt (hunk 34 above)
Comment (AlexisN94) [reply app/src/main/java/com/alexis/myanimecompanion/ui/profile/ProfileViewModel.kt:59]: was fixed in base branch

File: app/src/main/java/com/alexis/myanimecompanion/ui/BindingAdapters.kt (hunk 33 above)
Comment (AlexisN94) [reply app/src/main/java/com/alexis/myanimecompanion/ui/BindingAdapters.kt]: If I understood correctly, that would defeat the purpose of using a BindingAdapter; that's what android:visibility does.
I thought about this as well and it occurred to me to check the id of the Group (it's either _group_not_logged_in_ or _group_logged_in_) and set visibility accordingly, but it didn't feel right to tie the adapter with those ids. If you think it's okay though, I can make that change.

Comment (andre-exd) [reply app/src/main/java/com/alexis/myanimecompanion/ui/BindingAdapters.kt]: The usage of this bindingAdapter would be almost as you had in the beginning:
android:visible="@{viewModel.user != null &amp;&amp; !viewModel.loading}".
In my understanding this should allow to be reused on situations where the boolean value can be validated using a value from viewModel.
Something as: android:visible="@{viewModel.showLoading}"

Comment (AlexisN94) [reply app/src/main/java/com/alexis/myanimecompanion/ui/BindingAdapters.kt]: Please check if it's okay now

**Review by andre-exd — APPROVED** [review]

File: app/src/main/java/com/alexis/myanimecompanion/ui/search/SearchFragment.kt [hunk 36]
```diff
@@ -32,15 +33,15 @@ class SearchFragment : Fragment(), SearchListAdapter.ClickListener {
...
+            viewModel = this@SearchFragment.viewModel
             rvSearchResultList.apply {
                 this.adapter = adapter
                 layoutManager = GridLayoutManager(requireContext(), 3)
             }
             etSearchQuery.setOnKeyListener(object : View.OnKeyListener {
                 override fun onKey(v: View?, keyCode: Int, event: KeyEvent?): Boolean {
                     if (keyCode == KeyEvent.KEYCODE_ENTER) {
-                        viewModel.search()
+                        this@SearchFragment.viewModel.search()
```
Comment (andre-exd) [inline app/src/main/java/com/alexis/myanimecompanion/ui/search/SearchFragment.kt:44]: This doesn't seem correct

**Review by andre-exd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/alexis/myanimecompanion/ui/search/SearchFragment.kt (hunk 36 above)
Comment (AlexisN94) [reply app/src/main/java/com/alexis/myanimecompanion/ui/search/SearchFragment.kt:44]: I don't think it's incorrect, just redundant perhaps, considering that at this point _this@SearchFragment.viewModel_ should be equal to _this@apply.viewModel_ (i.e. _viewModel_). Maybe I'm missing your point?

Comment (andre-exd) [reply app/src/main/java/com/alexis/myanimecompanion/ui/search/SearchFragment.kt:44]: I see

**Review by andre-exd — APPROVED** [review]

**Pull request merged**